homeassistant/components/environment_canada/sensor.py

```diff
@@ -49,107 +51,103 @@ def validate_station(station):
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Environment Canada sensor."""
 
     if config.get(CONF_STATION):
-        ec_data = ECData(
+        ec_data = ECWeather(
```

Comment:
We want libraries that use aiohttp client session to accept an optional session parameter so we can use the home assistant client session across the whole app. It would be good if the library can be updated to support this.

Here's the helper to get the home assistant client session:
https://github.com/home-assistant/core/blob/6e74f9013642edfc26bfbc68f08504f65d33d357/homeassistant/helpers/aiohttp_client.py#L29-L34